server/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunctionBuilder.java

```diff
@@ -112,6 +104,21 @@ protected DecayFunctionBuilder(String fieldName, BytesReference functionBytes) {
         this.functionBytes = functionBytes;
     }
 
+    /**
+    * Validate properties, override this function if you have different validation rules per score function
+    */
+    protected void validateProperties(String fieldName, Object scale, double decay){
```

Comment:
This could simply be a `validateDecay` function. The null checks will always be universal.

But, I think this idea of a protected/Overridable function is a good one.

Reply:
@benwtrent Thanks for the comments! I'll work on it in the weekend